CMakeLists.txt

```diff
@@ -311,33 +359,34 @@ set(CMAKE_CXX_FLAGS_SANITIZE
     CACHE STRING "Flags used by the C++ compiler during santizer debug builds.")
 
 enable_language(CXX)
-
+find_package(Threads REQUIRED)
```

Comment:
seems like an arbitrary place to have it between the CXX and C flags. How about moving that down a bit .